test/server/GraknRunnerBase.java

```diff
@@ -146,6 +154,34 @@ public void start() {
         }
     }
 
+    private CountDownLatch checkServerStarted() {
+        CountDownLatch latch = new CountDownLatch(1);
+        Timer timer = new Timer();
+        timer.scheduleAtFixedRate(new TimerTask() {
+            int retryNumber = 0;
+            @Override
+            public void run() {
+                retryNumber++;
+                if (retryNumber % 4 == 0) {
+                    System.out.println(String.format("Waiting for %s server to start (%ds)...",
+                            name(), retryNumber * SERVER_ALIVE_POLL_INTERVAL_MILLIS / 1000));
+                }
+                String lsof;
+                try {
+                    lsof = executor.command("lsof", "-i", ":" + port).readOutput(true).execute().outputString();
+                } catch (IOException | InterruptedException | TimeoutException e) {
+                    lsof = "";
+                }
+                if (lsof != null && !lsof.isEmpty()) {
+                    latch.countDown();
+                    timer.cancel();
+                }
+                if (retryNumber > SERVER_ALIVE_POLL_MAX_RETRIES) timer.cancel();
+            }
+        }, 0, 500);
+        return latch;
+    }
```

Comment:
I actually considered simply using a `while` loop with a `Thread.sleep`; however, this was met with a stern warning from IntelliJ that it's bad practice. I found some relevant articles on the issue: https://josephmate.wordpress.com/2016/02/04/how-to-avoid-busy-waiting/

The current implementation is wordy, but, the actual _usage_ of `checkServerStarted` is the elegant one-liner
```java
boolean started = checkServerStarted().await(SERVER_STARTUP_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS);
```